src/odemis/driver/test/pwrmccdaq_test.py

```diff
@@ -0,0 +1,322 @@
+        Also does a small check for the poll thread to not have started when there are no status
+        DI's to check.
+        """
+        # test for an invalid DI channel number
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = {12: ["interlockTriggered", False], 99: ["testProperty", True]}
+        with self.assertRaises(ValueError):
+            pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        # test for an invalid structure of the channel dict
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = {11: [True, "testProperty"]}
+        with self.assertRaises(TypeError):
+            pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        # check list with no channel values given
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = {}
+        mcc_device = pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+
+        # check if there are no extra Boolean VA's
+        for va in model.getVAs(self.mcc_device):
+            self.assertFalse(isinstance(va, model.BooleanVA))
+
+        # be sure that there is no poll tread started as there are no Boolean VA's to check
+        self.assertFalse(mcc_device._status_thread)
+
+    @tag('skip_setup')
+    def test_multiple_di_channels(self):
+        """
+        Check if using multiple DI channels divided over both ports will
+        result in accurate status bit checks and VA changes.
+        """
+        MOD_CONFIG_MCC = copy.deepcopy(CONFIG_MCC)
+        MOD_CONFIG_MCC["di_channels"] = {
+            1: ["testPortABit1", False], 2: ["testPortABit2", False], 3: ["testPortABit3", False],
+            4: ["testPortABit4", False], 8: ["testPortBBit0", False], 9: ["testPortBBit1", False],
+            10: ["testPortBBit2", False], 11: ["testPortBBit3", False], 12: ["testPortBBit4", False],
+            13: ["testPortBBit5", False], 14: ["testPortBBit6", False], 15: ["testPortBBit7", False],
+        }
+
+        mcc_device = pwrmccdaq.MCCDevice(**MOD_CONFIG_MCC)
+        va_list = []
+
+        for name, va in model.getVAs(mcc_device).items():
+            if name.startswith("testPort"):
+                va_list.append(va)
+
+        # all status VA's should be False, as configured
+        for comp_va, key in zip(va_list, MOD_CONFIG_MCC["di_channels"].keys()):
+            self.assertEqual(comp_va.value, MOD_CONFIG_MCC["di_channels"][key][1])
+
+        # set all DI ports to HIGH
+        mcc_device.device.DConfig(0x01, 0xff)
+        mcc_device.device.DConfig(0x04, 0xff)
+        time.sleep(0.15)  # wait a bit to update the port bit status
+
+        # trigger bits 2 and 3 for port A -> binary 0b11110011 -> hex 0x4d
+        mcc_device.device.DOut(0x01, 0xf3)
+
+        # trigger bits 1, 4, 5 and 7 for port B -> binary 0b01001101 -> hex 0x4d
+        mcc_device.device.DOut(0x04, 0x4d)
+        time.sleep(0.15)  # wait a bit to update the port bit status
```

Comment:
Does this code work on a real device or just the simulator? Is just the simulator, indicate so. 

I am a little bit confused by the usage of DConfig. I thought it's to indicate which port is used for DO or DI. It should be done by the driver, right?